internal/clients/grpc/connect.go

```diff
@@ -45,20 +50,52 @@ func NewClientHandler(c *Client) ClientHandler {
 
 // Connect creates new gRPC client and connect to gRPC server.
 func Connect(cfg Config) (*gogrpc.ClientConn, bool, error) {
-	var opts []gogrpc.DialOption
+	opts := []gogrpc.DialOption{
+		gogrpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor()),
+	}
 	secure := false
 	tc := insecure.NewCredentials()
 
-	if cfg.ClientTLS && cfg.CACerts != "" {
-		var err error
-		tc, err = credentials.NewClientTLSFromFile(cfg.CACerts, "")
+	if cfg.ServerCAFile != "" {
+		tlsConfig := &tls.Config{}
+
+		// Loading System certificates
+		sysCertPool, err := x509.SystemCertPool()
```

Comment:
can `x509.SystemCertPool()` return an empty `sysCertPool` together with a `nil` error? If not I suggest this refactor

```go
		// Loading System certificates
		sysCertPool, err := x509.SystemCertPool()
		if err != nil {
			return nil, secure, fmt.Errorf("failed to load system cert pool: %w", err)
		}
		tlsConfig.RootCAs = sysCertPool
```

Reply:
@rodneyosodo , System Certs is removed in this PR, It will be added in separate   PR